### ISSUE TITLE
`Cannot read properties of undefined` thrown when an error occurred in an integrated workflow

### ISSUE THREAD

**Issue by d3rwan** [n8n-io/n8n#2757]
**Describe the bug**
When executing a sub-worflow thanks to `executeWorkflow` command, if the sub-workflow throw an error then another error is catched by the parent workflow. 

```
TypeError: Cannot read properties of undefined (reading 'main')
      at Object.additionalData.executeWorkflow (../n8n/src/WorkflowRunnerProcess.ts:298:16)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at Object.execute (../n8n-nodes-base/nodes/ExecuteWorkflow/ExecuteWorkflow.node.ts:200:25)
      at ../n8n-core/src/WorkflowExecute.ts:814:26
```

The source of the `undefined` seems to be https://github.com/n8n-io/n8n/blob/master/packages/cli/src/WorkflowRunnerProcess.ts#L289 (there's no `data.main` in the returnData field when the node is in error I guess). 

**To Reproduce**
1. Create a first workflow with an error 
```
{"nodes":[{"parameters":{},"name":"Start","type":"n8n-nodes-base.start","typeVersion":1,"position":[240,300]},{"parameters":{"functionCode":"// Code here will run only once, no matter how many input items there are.\n// More info and help: https://docs.n8n.io/nodes/n8n-nodes-base.function\n\n// Loop over inputs and add a new field called 'myNewField' to the JSON of each one\nthrow new Error(\"test\");\n\n// You can write logs to the browser console\nconsole.log('Done!');\n\nreturn items;"},"name":"Function","type":"n8n-nodes-base.function","typeVersion":1,"position":[680,300]},{"parameters":{},"name":"NoOp","type":"n8n-nodes-base.noOp","typeVersion":1,"position":[900,300]},{"parameters":{"values":{"boolean":[{"name":"test1"}]},"options":{}},"name":"Set","type":"n8n-nodes-base.set","typeVersion":1,"position":[460,300]}],"connections":{"Start":{"main":[[{"node":"Set","type":"main","index":0}]]},"Function":{"main":[[{"node":"NoOp","type":"main","index":0}]]},"Set":{"main":[[{"node":"Function","type":"main","index":0}]]}}}
```
2. Create another workflow calling the first one
```
{"nodes":[{"parameters":{},"name":"Start","type":"n8n-nodes-base.start","typeVersion":1,"position":[240,300]},{"parameters":{"workflowId":"5"},"name":"Execute Workflow","type":"n8n-nodes-base.executeWorkflow","typeVersion":1,"position":[680,300]},{"parameters":{},"name":"NoOp","type":"n8n-nodes-base.noOp","typeVersion":1,"position":[900,300]},{"parameters":{"values":{"boolean":[{"name":"test2"}]},"options":{}},"name":"Set","type":"n8n-nodes-base.set","typeVersion":1,"position":[460,300]}],"connections":{"Start":{"main":[[{"node":"Set","type":"main","index":0}]]},"Execute Workflow":{"main":[[{"node":"NoOp","type":"main","index":0}]]},"Set":{"main":[[{"node":"Execute Workflow","type":"main","index":0}]]}}}
```

**Expected behavior**
Get the original thrown error in the parent workflow

**Environment (please complete the following information):**
 - OS : MacOS Monterrey 12.1
 - n8n Version : 0.161.1
 - Node.js Version : v16.13.0
 - Database system : Postgres (Docker image postgres:alpine3.14) 
 - Operation mode : own (n8n instance with custom n8n-nodes-module) 

**Additional context**

**Comment by janober** [comment]
Thanks a lot for making us aware of it and providing an easy way to reproduce it.

Issue got fixed and will be released with the next version.

**Comment by janober** [comment]
Fix got released with `n8n@0.162.0`
